apps/pie-storybook/stories/pie-tag.stories.ts

```diff
@@ -96,24 +100,66 @@ export default tagStoryMeta;
 const Template : TemplateFunction<TagProps> = ({
     variant,
     size,
-    isInteractive,
     isStrong,
-    disabled,
+    isDimmed,
+    isIconOnly,
+    hasLeadingIcon,
     showIcon,
     slot,
-    iconPlacement,
-}) => html`
-    <pie-tag
-        variant="${ifDefined(variant)}"
-        size="${ifDefined(size)}"
-        iconPlacement="${ifDefined(iconPlacement)}"
-        ?isInteractive="${isInteractive}"
-        ?isStrong="${isStrong}"
-        ?disabled="${disabled}">
-        ${showIcon ? html`<icon-heart-filled slot="icon"></icon-heart-filled>` : nothing}
-        ${sanitizeAndRenderHTML(slot)}
-    </pie-tag>
-`;
+}) => {
+    const hasText = slot && slot.trim() !== '';
+    const shouldBeIconOnly = showIcon && !hasText;
+    const shouldHaveLeadingIcon = showIcon && hasText;
+
+    // Check for incorrect property settings
+    let instructionalMessage = '';
+
+    if (showIcon) {
+        // When showIcon is enabled, check for missing attributes
+        if (shouldBeIconOnly && !isIconOnly) {
+            instructionalMessage = 'Set "isIconOnly" to true for icon-only tags.';
+        } else if (shouldHaveLeadingIcon && !hasLeadingIcon) {
+            instructionalMessage = 'Set "hasLeadingIcon" to true for tags with icon + text.';
+        } else if (!shouldBeIconOnly && isIconOnly) {
+            instructionalMessage = 'Set "isIconOnly" to false when tag has text content.';
+        } else if (!shouldHaveLeadingIcon && hasLeadingIcon) {
+            instructionalMessage = 'Set "hasLeadingIcon" to false for icon-only tags.';
+        }
+    }
+
+    // When showIcon is disabled, check for incorrectly set attributes
+    if (!showIcon && (isIconOnly || hasLeadingIcon)) {
+        if (isIconOnly) {
+            instructionalMessage = 'Set "isIconOnly" to false when no icon is provided.';
+        } else if (hasLeadingIcon) {
+            instructionalMessage = 'Set "hasLeadingIcon" to false when no icon is provided.';
+        }
+    }
+
+    return html`
+        ${instructionalMessage ? html`
+            <pie-notification
+                variant="warning"
+                position="inline-content"
+                .isOpen="${true}"
+                .isDismissible="${false}"
+                .isCompact="${true}">
```

Comment:
don't think we need the . here


```suggestion
                isOpen
                isDismissible
                isCompact
```